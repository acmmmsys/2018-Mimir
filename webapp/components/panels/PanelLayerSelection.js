import React from 'react';
import { connect } from 'react-redux';
import { compose, withProps, nest, withHandlers, lifecycle } from 'recompose';
import { requestCnnLayers, selectCnnLayer, requestImageVisualization } from 'actions';

import Panel from 'layout/Panel'
import TableSelector from 'components/ui/TableSelector';

const enhance = compose(
    connect(
        state => ({
            layers: state.cnn.layers,
            imageId: state.cnn.selectedImageId,
            layerId: state.cnn.selectedLayer,
            classId: state.cnn.selectedClass,
            selectedRow: state.cnn.layers.indexOf(state.cnn.selectedLayer),
            loading: state.loading.layers,
        }),
        dispatch => ({
            requestCnnLayers() {
                dispatch(requestCnnLayers());
            },
            selectCnnLayer(layerIndex) {
                dispatch(selectCnnLayer(layerIndex));
            },
            selectVisualizationLayer(imageId, layerId, classId){
                dispatch(requestImageVisualization(imageId, layerId, classId));
            },
        })
    ),
    withHandlers({
        onClick: props => rowIndex => {
            props.selectCnnLayer(rowIndex);
            props.selectVisualizationLayer(
                props.imageId,
                props.layers[rowIndex],
                props.classId
            );
        },
    }),
    withProps(props => ({
        isEmpty: !props.imageId,
        header: ['Layer Name'],
        content: props.layers.map(layerName =>  [layerName]),
        emptyMessage: 'The result of the file analysis will appear here',
    })),
    lifecycle({
        componentWillMount()Â {
            this.props.requestCnnLayers();     
        },
    })
);

export default enhance(nest(Panel, TableSelector));