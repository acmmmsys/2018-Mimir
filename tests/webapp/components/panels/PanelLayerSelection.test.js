import React from 'react';
import PanelLayerSelection from '../../../../webapp/components/panels/PanelLayerSelection'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

let store;
let wrapper;

test('PanelLayerSelection does render', () => {
  const wrapper = shallow(
      <PanelLayerSelection store={mockStore(initialState)}>This is a test!</PanelLayerSelection>
  );
  expect(wrapper).toMatchSnapshot();
});
