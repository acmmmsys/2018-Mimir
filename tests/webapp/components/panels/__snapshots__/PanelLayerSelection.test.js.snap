// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelLayerSelection does render 1`] = `
<withHandlers(withProps(lifecycle(nest(defaultProps(branch(branch(Panel))), TableSelector))))
  classId={6}
  layerId="block5_conv4"
  layers={Array []}
  loading={false}
  requestCnnLayers={[Function]}
  selectCnnLayer={[Function]}
  selectVisualizationLayer={[Function]}
  selectedRow={-1}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
>
  This is a test!
</withHandlers(withProps(lifecycle(nest(defaultProps(branch(branch(Panel))), TableSelector))))>
`;
